synr/compiler.py

```diff
@@ -273,22 +279,25 @@ def compile_stmt(self, stmt: py_ast.stmt) -> Stmt:
                     self.span_from_asts(stmt.items),
                 )
             wth = stmt.items[0]
-            lhs_var: Optional[Var]
             if wth.optional_vars:
                 l = self.compile_expr(wth.optional_vars)
-                if not isinstance(l, Var):
+                if isinstance(l, Var):
+                    lhs_vars = [l]
+                elif (isinstance(l, ArrayLiteral) or isinstance(l, Tuple)) and not [
```

Comment:
Can you move the check that all elements are vars inside of this statement and report a specialized error message there.